lib/spack/spack/environment/environment.py

```diff
@@ -519,6 +519,13 @@ def regenerate(self, concretized_specs):
             tty.debug("View at %s does not need regeneration." % self.root)
             return
 
+        # Check if a folder is blocking us from creating a new view directory
+        if os.path.lexists(new_root):
+            tty.warn("Not updating view because {0} already exists. To update the "
```

Comment:
As discussed on slack, generating a view is part of the expected behavior of `spack -e . install`, so this should be a fatal error (whether through `tty.die` or `raise`). Also since we don't have to support python 2.6, this can be `{}` instead of `{0}` :)

Reply:
Fixed that